### DISCUSSION TITLE
Any thoughts/idea on implementing multitenancy/team using Prisma Adapter ?

### DISCUSSION THREAD

**Discussion by octalpixel** [nextauthjs/next-auth#3950] [Ideas]
### Question 💬

How possible is it to implement a data modeling highlighted on https://blitzjs.com/docs/multitenancy and https://blog.bullettrain.co/teams-should-be-an-mvp-feature/  using the Prisma adapter?

**_Snippet from the Blitz documentation_**
- The Organization is the "God" model which owns everything for an account
- An Organization has many Users through Membership
- Every other model in the system has an organizationId to indicate who owns it.
- A User can have access to multiple Organizations
- When assigning an entity to a user, like a task, assign the task to the user's Membership instead of directly to the user.

The general idea is to either use events https://next-auth.js.org/configuration/events or modify the existing Prisma provider to create the organization.

Can someone share some insights on creating something similar using next-auth.

### How to reproduce ☕️

-

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
As a side comment, if we are going to look into this, it has to be database/adapter agnostic.

**Comment by ananni13** [comment]
I don't know if this should be in NextAuth, seems kind of out of scope. You use NextAuth to authenticate your `User`, not a `Membership` or `Organization`. Implementing and managing multitenancy is then up to your app architecture/logic.

You can just merge the schema needed [from NextAuth](https://next-auth.js.org/adapters/prisma#create-the-prisma-schema) with the one suggested by [Blitz for multitenancy](https://blitzjs.com/docs/multitenancy#data-model) and you're good to go. Your `User` will authenticate and have his own session, then handle creating/switching `Organization`s (and related `Membership`s) in the app.
